File: tests/fixtures/zig/production_api_service.zig
const std = @import("std");

const AppConfig = struct {
    host: []const u8,
    port: u16,
    database_url: []const u8,
    redis_url: []const u8,
    jwt_secret: []const u8,
    log_level: []const u8,
    max_connections: u32,
    timeout_secs: u64,
    retry_count: u32,
    cache_ttl: u64,
    rate_limit: u32,
};

fn processEvent(data: []const u8) i32 {
    const a = data[0];
    const b = data[1];
    const c = data[2];
    const d = data[3];
    const e = data[4];
    const f = data[5];
    const g = data[6];
    const h = data[7];
    if (a > 100) {
        return -1;
    }
    if (b > 200) {
        return -2;
    }
    const result = @as(i32, a) + @as(i32, b) + @as(i32, c) + @as(i32, d);
    const extra = @as(i32, e) + @as(i32, f) + @as(i32, g) + @as(i32, h);
    return result + extra;
}

fn cleanHelper(x: i32, y: i32) i32 {
    return x + y;
}

File: tests/fixtures/rust/production_api_service.rs
/// Realistic Rust API service with intentional smells for testing.

struct AppConfig {
    host: String,
    port: u16,
    database_url: String,
    redis_url: String,
    jwt_secret: String,
    cors_origins: Vec<String>,
    log_level: String,
    max_connections: u32,
    timeout_secs: u64,
    retry_count: u32,
    cache_ttl: u64,
    rate_limit: u32,
    upload_dir: String,
}

fn dispatch_command(cmd: &str) -> &str {
    match cmd {
        "start" => "starting",
        "stop" => "stopping",
        "restart" => "restarting",
        "status" => "checking",
        "deploy" => "deploying",
        "rollback" => "rolling back",
        "migrate" => "migrating",
        "seed" => "seeding",
        _ => "unknown",
    }
}

fn process_event(data: &[u8]) -> i32 {
    let a = data.get(0).copied().unwrap_or(0);
    let b = data.get(1).copied().unwrap_or(0);
    let c = data.get(2).copied().unwrap_or(0);
    let d = data.get(3).copied().unwrap_or(0);
    let e = data.get(4).copied().unwrap_or(0);
    let f = data.get(5).copied().unwrap_or(0);
    let g = data.get(6).copied().unwrap_or(0);
    let h = data.get(7).copied().unwrap_or(0);
    if a > 100 {
        return -1;
    }
    if b > 200 {
        return -2;
    }
    let result = a as i32 + b as i32 + c as i32 + d as i32;
    let extra = e as i32 + f as i32 + g as i32 + h as i32;
    result + extra
}

fn clean_helper(x: i32, y: i32) -> i32 {
    x + y
}
